docs/operating/compactor-backlog.md

```diff
@@ -0,0 +1,55 @@
+# Troubleshoot Compactor Backlog
+
+Compactor is one of the most important components in Thanos. It is responsible for doing compaction, downsampling and retention for the data in the object storage.
+
+When your system contains a lot of block producers (Sidecar, Rule, Receiver, etc) or the scale is large, compactor might not be able to keep up with the data producing rate and it falls behind, which causes a lot of backlogged work. This document will help you to troubleshoot the backlog compaction issue and how to scale the compactor.
+
+## Detect the backlog
+
+Self monitoring for the monitoring system is important. We highly recommend you to set up the Thanos Grafana dashboards and alerts to monitor the thanos components. Without self monitoring, it is hard to detect the issue and fix the problems.
+
+If you find these issues in your own Thanos deployment, then your compactor might be backlogged and needs to scale more:
+1. The long-term query in your Grafana dashboard is much slower than usual.
+2. The compactor stops downsampling blocks and `thanos_compact_downsample_total` metric doesn't increase.
+3. The compactor stops retention. If the retention period is set to 1 month but you can still see older blocks from the bucket UI then that's the case.
+4. `thanos_compact_iterations_total` metric doesn't increase or the rate is very low.
+
+Current implementation, compactor will perform compaction, downsampling and retention phases in order, which means that if the compaction work is not finished, downsampling and retention phase won't start. So that's why you will find the symptom 2 and 3 mentioned above.
```

Comment:
```suggestion
In the current implementation, the compactor will perform compaction, downsampling, and retention phases in order, which means that if the compaction work is not finished, the downsampling and retention phase won't start. So that's why you will find symptoms 2 and 3 mentioned above.
```